### PR TITLE
Added support for more DNS record Types and TTL parameter

### PR DESCRIPTION
**Pull request by xtof-osd**
Building on a previous PR (https://github.com/synthesio/infra-ovh-ansible-module/pull/31),
as suggested then, added full support for any DNS record supported by OVH API.

### CODE REVIEW

File: plugins/modules/domain.py [hunk 1]
```diff
@@ -28,10 +28,20 @@
         description: The domain to modify
     record_type:
         required: false
-        description: The DNS record type (A, CNAME, TXT, AAAA, NS, SRV, MX, , DKIM, SPF)
+        description: The DNS record type, any of the type supported by OVH API
+        default: A
     state:
         required: false
-        description: The state
+        description:  Indicate the desired state for the record
```
Comment (tortuegenialez) [inline plugins/modules/domain.py:35]: Double space before *Indicate*

File: plugins/modules/domain.py [hunk 2]
```diff
@@ -62,9 +73,17 @@ def run_module():
...
-    ))
+        record_type=dict(choices=['A', 'AAAA', 'CAA',
+                                  'CNAME', 'DKIM', 'DMARC',
+                                  'DNAME', 'LOC', 'MX',
+                                  'NAPTR', 'NS', 'PTR',
+                                  'SPF', 'SRV', 'SSHFP',
+                                  'TLSA', 'TXT'],
+                         default='A'),
+        state=dict(choices=['present', 'absent'], default='present'),
+        record_ttl=dict(type='int',required=False, default=0)
```
Comment (tortuegenialez) [inline plugins/modules/domain.py:84]: The default should be null or empty string to let the API handle the default.

**Review by tortuegenialez — CHANGES_REQUESTED** [review]
The default ttl should be left to the API

File: plugins/modules/domain.py (hunk 2 above)
Comment (xtof-osd) [reply plugins/modules/domain.py:84]: Do not agree, the API does not explictly specify the default it uses ( which eventually is 0 when you use the API, and 600 if you use OVH API). 
I think specifying an explicit default value is far better than implicit default value that might change.

File: plugins/modules/domain.py (hunk 1 above)
Comment (xtof-osd) [reply plugins/modules/domain.py]: Fixed, Thanks.

File: plugins/modules/domain.py (hunk 2 above)
Comment (Skunnyk) [reply plugins/modules/domain.py:84]: I don't have a strong advice on this topic.
"0"' means "use the zone TTL", and I think this is a good default

File: plugins/modules/domain.py [hunk 3]
```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 
-from __future__ import (absolute_import, division, print_function)
+from __future__ import absolute_import, division, print_function
```
Comment (Skunnyk) [inline plugins/modules/domain.py:4]: Please do not add syntaxes changes. The syntax is based on what is recommended on ansible documentation: https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html#creating-an-info-or-a-facts-module

**Comment by Skunnyk** [comment]
Merged in 6bb63f82e1636082cf99737daddc5a9af087bdb5 thank you ! See #47

**Pull request closed without merging**